Bottom: f5c3c4cecac2626a659f3965528865b0340443a0
Top:    7fe5b6815e28461fa033e596c8d29b8717a40bb8
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-10 16:58:02 -0500

Refresh of add-buildbot-role-and-new

---
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index fa19756e..cd939d11 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -1,52 +1,97 @@
 ---
 # defaults file for buildbot
 
-# The hostname passed as an envirnment variable into the container
-buildbot_ip_addr: '127.0.0.1'
-buildbot_hostname: buildbot.example.com
-buildbot_port_args: 
-  - '8080'
-  - '8443'
-  - '9989'
+# host containing build assets
+asset_host: 'http://{{ansible_facts.default_ipv4.address}}/'
 
-# Network names
-buildbot_networks:
-  - name: '{{ docker_network_frontend }}'
+# backup host directory
+backup_hostdir: '{{ docker_backup_dir }}/buildbot'
 
-buildbot_data_path: '/buildbot'
+# numeric ID of buildbot user
+buildbot_uid: 1000
 
-# the name of the image being duilt and used for the container
-buildbot_image_repo: https://github.com/novatechweb/docker-buildbot-master.git
-buildbot_image_name: '{{ docker_registry_username }}/buildbot'
-buildbot_image_dir: '{{ docker_projects_dir }}/buildbot-master'
+# numeric version of buildbot service
+buildbot_version: '1.1.2'
 
-# the name of the container being started
-buildbot_container_name: '{{ docker_name_prefix }}buildbot'
+# port number for workers to communicate with master
+buildbot_worker_port: 9989
 
-# the name of the data-volume used by the container
-buildbot_dv_name: '{{ docker_name_prefix }}buildbot_DV'
+# port number for unencrypted web access
+buildbot_http_port: 8080
 
-# restore directories to temporarly store data being restored into docker containers
-buildbot_docker_backup_dir: '{{ docker_backup_dir }}/buildbot'
-host_buildbot_docker_restore_dir: '{{ bacula_dest }}{{ buildbot_docker_backup_dir }}'
+# port number for encrypted web access
+buildbot_https_port: 8443
 
-buildbot_build_args:
-  BUILDBOT_UID: '1000'
-  BUILDBOT_VERSION: 'v1.1.2'
-  BUILDBOT_TIMEZONE: 'America/Chicago'
+# The hostname passed as an envirnment variable into the container
+buildbot_ip_addr: '127.0.0.1'
+buildbot_hostname: buildbot.example.com
+buildbot_port_args: 
+  - '{{buildbot_http_port}}'
+  - '{{buildbot_https_port}}'
+  - '{{buildbot_worker_port}}'
 
-buildbot_volumes:
-  - '{{ buildbot_dv_name }}:{{ buildbot_data_path }}:z'
+# host path containing configuration files
+config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
 
-buildbot_env:
-  BUILDBOT_DATABASE: 'sqlite:///{{ buildbot_data_path }}/state.sqlite'
-  BUILDBOT_WEB_PORT: '8080'
+# environment passed to the container
+container_env:
+  BUILDBOT_KNOWN_HOSTS_FILE: '{{secrets_path}}/{{secrets_known_hosts}}'
+  BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
+  BUILDBOT_WEB_PORT: '{{buildbot_http_port}}'
   BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
   BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
   BUILDBOT_SMTP_USER: 'Dean.cox'
   BUILDBOT_SMTP_PASS: 'no5156'
-  BUILDBOT_WORKER_PORT: '9989'
+  BUILDBOT_WORKER_PORT: '{{buildbot_worker_port}}'
   BUILDBOT_WORKER_NTEL: 'worker-ntel'
   BUILDBOT_WORKER_NTEL_PASS: '{{ buildbot_worker_ntel_password }}'
   BUILDBOT_WORKER_PTXDIST: 'worker-ptxdist'
   BUILDBOT_WORKER_PTXDIST_PASS: '{{ buildbot_worker_ptxdist_password }}'
+
+# the name of the container being started
+container_name: '{{ docker_name_prefix }}buildbot'
+
+# networks of the container
+container_networks:
+  - name: '{{ docker_network_frontend }}'
+
+# volumes mounted within the container
+container_volumes:
+  - '{{ data_volume }}:{{ data_path }}:z'
+  - '{{ config_hostdir }}:{{ secrets_path }}'
+
+# mount path of the data volume
+data_path: '/buildbot'
+
+# name of the data volume
+data_volume: '{{ docker_name_prefix }}buildbot_DV'
+
+# hostname
+hostname: buildbot-worker-ntel
+
+# build arguments of the image
+image_args:
+  BUILDBOT_UID: '{{buildbot_uid}}'
+  BUILDBOT_VERSION: 'v{{buildbot_version}}'
+  BUILDBOT_TIMEZONE: '{{ansible_facts.date_time.tz}}'
+
+# directory of the image source
+image_dir: '{{ docker_projects_dir }}/buildbot-master'
+
+# name of the image being built
+image_name: '{{ docker_registry_username }}/buildbot'
+
+# repository URI of the image source
+image_repo: https://github.com/novatechweb/docker-buildbot-master.git
+
+# restoration host directory
+restore_hostdir: '{{ bacula_dest }}{{ backup_hostdir }}'
+
+# host path containing secrets
+secrets_hostdir: '{{ config_hostdir }}'
+
+# name of the ssh known_hosts secret
+secrets_known_hosts: 'known_hosts'
+
+# mount path of the secrets volume
+secrets_path: '/run/secrets'
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index 3415bab6..63fac665 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -8,30 +8,31 @@
 
 - name: Create buildbot data volume
   docker_volume:
-    name: '{{ buildbot_dv_name }}'
+    name: '{{ data_volume }}'
 
 - name: Checkout image repo
   git:
-    repo: '{{ buildbot_image_repo }}'
+    repo: '{{ image_repo }}'
     version: master
-    dest: '{{ buildbot_image_dir }}'
+    dest: '{{ image_dir }}'
 
 - name: Create buildbot master image
   docker_image:
-    name: '{{ buildbot_image_name }}'
+    name: '{{ image_name }}'
     tag: '{{ docker_image_tag }}'
-    path: '{{ buildbot_image_dir }}'
-    buildargs: '{{ buildbot_build_args }}'
+    path: '{{ image_dir }}'
+    buildargs: '{{ image_args }}'
     force: true
 
 - name: Start buildbot container
   docker_container:
-    name: '{{ buildbot_container_name }}'
-    image: '{{ buildbot_image_name }}:{{ docker_image_tag }}'
-    networks: '{{ buildbot_networks }}'
-    volumes: '{{ buildbot_volumes }}'
-    env: '{{ buildbot_env }}'
+    name: '{{ container_name }}'
+    image: '{{ image_name }}:{{ docker_image_tag }}'
+    networks: '{{ container_networks }}'
+    volumes: '{{ container_volumes }}'
+    env: '{{ container_env }}'
     ports: '{{ buildbot_port_args }}'
+    state: started
 
 - name: Add buildbot_master host
   add_host:
